src/main/java/ml/compute/InputData.java

```diff
@@ -0,0 +1,28 @@
+package ml.compute;
+
+import java.util.Set;
+
+import ml.preprocessor.types.PreprocessedData;
+import ml.preprocessor.types.RawData;
+
+/**
+ * Created by mikeecb on 17/10/2015.
+ */
+public class InputData {
```

Comment:
I agree with Timotej, the InputData that we are passing the HeartRateComputer is going to be a rich DAO (it will probably contain, audio, video features).

With a visitor pattern we can extract all of these features at the time of visit, whereas having an interface passing the data into the HRComputer increases coupling by requiring the HRComputer to extract the data itself, or require another class to to do it.